packages/compass-aggregations/src/modules/aggregation.ts

```diff
@@ -204,9 +225,89 @@ const reducer: Reducer<State> = (state = INITIAL_STATE, action) => {
   return state;
 };
 
+const confirmWriteOperationIfNeeded = async (
+  instance: MongoDBInstance,
+  dataService: DataService,
+  namespace: string,
+  pipeline: Document[]
+) => {
+  if (!pipeline.length) {
+    return true;
+  }
+
+  const lastStageOperator = getStageOperator(pipeline[pipeline.length - 1]);
+  let typeOfWrite;
+
+  if (!lastStageOperator || !isOutputStage(lastStageOperator)) {
+    return true;
+  }
+
+  const destinationNamespace = getDestinationNamespaceFromStage(namespace, {
+    [lastStageOperator]: pipeline[pipeline.length - 1][lastStageOperator],
+  });
+
+  if (lastStageOperator === '$out') {
+    let isOverwritingCollection;
```

Comment:
To avoid this being of type any, you probably want to indicate what type you expect to assign here
```suggestion
    let isOverwritingCollection: boolean;
```